### PR TITLE
creating a new VelocityModeHandler

### PR DESCRIPTION
**Pull request by freddyucv**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityAdminMode.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.liferay.portal.model.User;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+/**
+ * Velocity renderer for PageMode#ADMIN_MODE
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityAdminMode.java]: it seems you are introducing a new concept, explain here a bit more about the intention

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityAdminMode.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ */
+public class VelocityAdminMode extends VelocityLiveMode {
+
+    public VelocityAdminMode(HttpServletRequest request, HttpServletResponse response, String uri, Host host) {
+        super(request, response, uri, host);
+    }
+
+    User getUser() {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityAdminMode.java]: it seems you are overriding a method, if so use override annotation!

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityLiveMode.java [hunk 3]
```diff
@@ -185,4 +180,15 @@ public final void serve(final OutputStream out) throws DotDataException, IOExcep
         }
     }
 
+    User getUser() {
+        User user = null;
+        HttpSession session = request.getSession(false);
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityLiveMode.java]: set to final

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityModeHandler.java [hunk 4]
```diff
@@ -39,7 +39,7 @@
             .put(PageMode.PREVIEW_MODE, VelocityPreviewMode::new)
             .put(PageMode.EDIT_MODE, VelocityEditMode::new)
             .put(PageMode.LIVE, VelocityLiveMode::new)
-            .put(PageMode.ADMIN_MODE, VelocityLiveMode::new)
+            .put(PageMode.ADMIN_MODE, VelocityAdminMode::new)
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityModeHandler.java]: Nice

**Review by jdotcms — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

**Review by jgambarios — APPROVED** [review]

**Pull request merged**